src/JNI.jl

```diff
@@ -87,518 +93,596 @@ const JNI_EINVAL       = convert(Cint, -6)              #/* invalid arguments */
 PtrIsCopy = Union{Ptr{jboolean},Ref{jboolean},Array{jboolean,}}
 AnyString = Union{AbstractString,Cstring,Ptr{UInt8}}
 
+struct JNIError <: Exception
+    msg::String
+end
+
+struct JavaVMOption
+    optionString::Ptr{UInt8}
+    extraInfo::Ptr{Nothing}
+end
+
+struct JavaVMInitArgs
+    version::Cint
+    nOptions::Cint
+    options::Ptr{JavaVMOption}
+    ignoreUnrecognized::Cchar
+end
+
 function load_jni(penv::Ptr{JNIEnv})
-    jnienv = unsafe_load(penv)
+    global jnienv = unsafe_load(penv)
     jniref[] = unsafe_load(jnienv.JNINativeInterface_) #The JNI Function table
     global jnifunc = jniref[]
```

Comment:
@aviks `jnifunc` is set here.

I also found that storing the `jnienv` as a global somehow makes x86 Windows work more smoothly. With that we have successful `JavaCall.init()` on x86 Windows. I wanted to see if I could understand why that makes a difference before I start cleaning up the globals.